### PR TITLE
clean up use of build runner internals

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Replace usage of build_runner internals with build_daemon shared by web_fs. Adds some workaround copying to keep tests running correctly.

https://github.com/flutter/flutter/issues/31803

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 1]
```diff
@@ -7,205 +7,72 @@ import 'dart:async';
...
+      includeTests: true,
+    );
+    client.startBuild();
+    bool success = true;
+    await for (BuildResults results in client.buildResults) {
+      final BuildResult result = results.results.firstWhere((BuildResult result) {
+        return result.target == 'web';
+      });
+      if (result.status == BuildStatus.failed) {
+        success = false;
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Should it break after this?

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 2]
```diff
@@ -7,205 +7,72 @@ import 'dart:async';
...
+    }
+    if (success && testOutputDir != null) {
+      final Directory rootDirectory = projectDirectory
+        .childDirectory('.dart_tool')
+        .childDirectory('build')
+        .childDirectory('flutter_web');
+
+      for (Directory childDirectory in rootDirectory
+        .listSync()
+        .whereType<Directory>()) {
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Maybe extract the iterable into a variable?

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 3]
```diff
@@ -7,205 +7,72 @@ import 'dart:async';
...
+        copyDirectorySync(childDirectory.childDirectory('lib'),
+          fs.directory(fs.path.join(testOutputDir, 'packages',
+          fs.path.basename(childDirectory.path))));
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: I think `dartfmt` would produce better code with a trailing comma.

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 4]
```diff
@@ -7,205 +7,72 @@ import 'dart:async';
...
       }
+      copyDirectorySync(rootDirectory
+        .childDirectory(projectName)
+        .childDirectory('test'),
+        fs.directory(fs.path.join(testOutputDir)));
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: a trailing comma might help here too

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Reformatted

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Reformatted

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Reformatted

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Fixed

**Pull request merged**